src/lib/completion.ts

```diff
@@ -1,131 +1,144 @@
-import { Document } from 'langchain/document';
-
+//https://js.langchain.com/docs/modules/agents/how_to/custom_llm_agent
 import { ChatOpenAI } from 'langchain/chat_models/openai';
 import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
-import { SystemMessage, BaseMessage, AIMessage, HumanMessage } from 'langchain/schema';
-
-import { ChatHistoryType, type ChatHistory } from '$lib/history';
-import { BytesOutputParser } from 'langchain/schema/output_parser';
-
-import { QdrantClient } from '@qdrant/js-client-rest';
+import {
+	type AgentAction,
+	type AgentFinish,
+	HumanMessage,
+	type InputValues,
+	type AgentStep,
+	AIMessageChunk
+} from 'langchain/schema';
+
+import { type ChatHistory } from '$lib/history';
+import { formatLogToString } from 'langchain/agents/format_scratchpad/log';
+import { RunnableSequence } from 'langchain/schema/runnable';
+import { AgentExecutor } from 'langchain/agents';
+import { PREFIX, SUFFIX, TOOL_INSTRUCTIONS_TEMPLATE } from './prompt';
+import { getTools } from './tools';
+import { colors } from './colors';
 
 const DEFAULT_MODEL = 'gpt-3.5-turbo';
-const DEFAULT_COLLECTION = 'default';
 
 export class ChatbotCompletion {
-	private model: ChatOpenAI;
 	private embeddings_model: OpenAIEmbeddings;
-
-	private qdrant_client: QdrantClient;
-	private qdrant_collection: string;
+	private executor: AgentExecutor | undefined;
+	private openai_api_key: string;
+	private model_name;
+	private verbose: boolean;
 
 	constructor(
 		openai_api_key: string,
+		verbose: boolean,
 		{
-			openai_model = DEFAULT_MODEL,
-			qdrant_collection = DEFAULT_COLLECTION
+			openai_model = DEFAULT_MODEL
 		}: {
 			openai_model?: string;
-			qdrant_collection?: string;
 		}
 	) {
-		this.model = new ChatOpenAI({
-			openAIApiKey: openai_api_key,
-			temperature: 0.7,
-			streaming: true,
-			maxTokens: 250,
-			modelName: openai_model,
-			verbose: true
-		});
+		this.verbose = verbose;
+		this.model_name = openai_model;
+		this.openai_api_key = openai_api_key;
 
 		this.embeddings_model = new OpenAIEmbeddings({
 			openAIApiKey: openai_api_key,
 			modelName: 'text-embedding-ada-002'
 		});
-		this.qdrant_client = new QdrantClient({
-			url: 'http://' + (process.env.QDRANT_HOST ?? 'localhost') + ':6333'
```

Comment:
We also need this so that it works with docker, as with docker compose the url will be qdrantdb

Reply:
This can be in a seperate PR